src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -179,11 +179,9 @@ private void AddErrorLabelIfRequired(Exception exception, SemanticVersion server
                     mongoException.AddErrorLabel("TransientTransactionError");
                 }
 
-                if ((exception is MongoConnectionException) || // network error
-                    (Feature.RetryableWrites.IsSupported(serverVersion) &&
-                    !Feature.ServerReturnsRetryableWriteErrorLabel.IsSupported(serverVersion)))
+                if (_command.Contains("txnNumber")) // operation is retryable
```

Comment:
See for details: https://github.com/mongodb/specifications/blob/49992a6d1c14b36421e778e78070a427887674fb/source/retryable-writes/retryable-writes.rst#why-does-the-driver-only-add-the-retryablewriteerror-label-to-errors-that-occur-on-a-mongoclient-with-retrywrites-set-to-true